starter-code/iteration-1/main.ts

```diff
@@ -1,18 +1,39 @@
 // Iteration 1
 // 1. Create an array of strings
 
+let toDoList: Array<string> = [];
+
 // 2. Create a function to add a task in to the array, call it `addTask`, the function has to accept a string and don't have to return anything.
 
+function addTask(task: string): Array<string> {
+  return toDoList.push(task);
+}
+
 // 3. Create a function to list all tasks, it must show in the console de task.
 
+function listAllTasks(): any {
+  for (var ix = 0; ix < toDoList.length; ix++) {
+    console.log(ix + 1 + ": " + toDoList[ix]);
+  }
+}
+
 // 4. Create a function to delete a task, you must find the task inside the array and delete it.
 
+function deleteTask(task: string): Array<string> {
+  let position = toDoList.indexOf(task);
+  if (position > -1) {
+    return toDoList.splice(position, 1);
```

Comment:
This would return the deleted element, you want to return the new length instead.